diff -urN ovito-v3.10.6/src/ovito/core/utilities/io/CompressedTextReader.cpp ovito-v3.10.6-modified-source/src/ovito/core/utilities/io/CompressedTextReader.cpp
--- ovito-v3.10.6/src/ovito/core/utilities/io/CompressedTextReader.cpp	2024-05-02 22:30:30.000000000 +0800
+++ ovito-v3.10.6-modified-source/src/ovito/core/utilities/io/CompressedTextReader.cpp	2024-09-08 01:37:28.904577400 +0800
@@ -63,7 +63,9 @@
             throw Exception(FileManager::tr("Failed to open input file: %1").arg(_uncompressor->errorString()));
         _stream = _uncompressor.get();
 #else
-        throw Exception(tr("Cannot open file '%1' for reading. This version of OVITO was built without I/O support for gzip compressed files."));
+        throw Exception(FileManager::tr(
+            "Cannot open file '%1' for reading. This version of OVITO was built without I/O support for gzip compressed files.")
+            .arg(_filename));
 #endif
     }
     else {
diff -urN ovito-v3.10.6/src/ovito/core/utilities/io/CompressedTextWriter.cpp ovito-v3.10.6-modified-source/src/ovito/core/utilities/io/CompressedTextWriter.cpp
--- ovito-v3.10.6/src/ovito/core/utilities/io/CompressedTextWriter.cpp	2024-05-02 22:30:30.000000000 +0800
+++ ovito-v3.10.6-modified-source/src/ovito/core/utilities/io/CompressedTextWriter.cpp	2024-09-08 01:37:16.716433500 +0800
@@ -50,7 +50,10 @@
             throw Exception(FileManager::tr("Failed to open output file '%1' for writing: %2").arg(_filename).arg(_compressor->errorString()));
         _stream = _compressor.get();
 #else
-        throw Exception(tr("Cannot open file '%1' for writing. This version of OVITO was built without I/O support for gzip compressed files.").arg(_filename));
+        throw Exception(
+            FileManager::tr(
+                "Cannot open file '%1' for writing. This version of OVITO was built without I/O support for gzip compressed files.")
+                .arg(_filename));
 #endif
     }
     else {
diff -urN ovito-v3.10.6/src/ovito/particles/import/gromacs/GroImporter.cpp ovito-v3.10.6-modified-source/src/ovito/particles/import/gromacs/GroImporter.cpp
--- ovito-v3.10.6/src/ovito/particles/import/gromacs/GroImporter.cpp	2024-05-02 22:30:30.000000000 +0800
+++ ovito-v3.10.6-modified-source/src/ovito/particles/import/gromacs/GroImporter.cpp	2024-09-08 20:42:15.058148100 +0800
@@ -71,8 +71,12 @@
     for(unsigned long long i = 0; i < 10; i++) {
         int i1, i2;
         char s1[6], s2[6];
-        if(sscanf(stream.readLine(), "%5i%5s%5s%5i", &i1, s1, s2, &i2) != 4 || i1 < 1 || i2 < 1 || qstrlen(stream.line()) < 20)
+        //if(sscanf(stream.readLine(), "%5i%5s%5s%5i", &i1, s1, s2, &i2) != 4 || i1 < 1 || i2 < 1 || qstrlen(stream.line()) < 20)
+        //    return false;
+        char s3[6], s4[6];
+        if(4 != sscanf(stream.readLine(), "%5c%5c%5c%5c", s1, s2, s3, s4) || qstrlen(stream.line()) < 20) {
             return false;
+        }
 
         // The following extra check is necessary to avoid mistaking an XDATCAR file, which stores the simulation cell
         // vectors in lines 3-5, for a GRO file (see test case 'POSCAR/XDATCAR.testcase'):
Binary files ovito-v3.10.6/src/ovito_exe/resources/ovito.aps and ovito-v3.10.6-modified-source/src/ovito_exe/resources/ovito.aps differ
